### PR TITLE
Update pulumi dependencies to v3.6.0

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ## HEAD (Unreleased)
 
+## 3.5.0 (June 30, 2021)
+
 - Update pulumi dependencies v3.5.1 (https://github.com/pulumi/pulumi-kubernetes/pull/1623)
```
Comment (viveklak) [inline CHANGELOG.md:5]: Since we are updating to 3.6.0 anyway
```suggestion
```

**Review by viveklak — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (lblackstone) [reply CHANGELOG.md:5]: I'd rather leave the link since the other PR also included SDK changes.

**Comment by lblackstone** [comment]
The NodeJS codegen has changed all of the `getResourceProperty` methods and broken a test. The change is from https://github.com/pulumi/pulumi/blame/2781cf03fee77ea09286d442ffe3a01e6823322c/pkg/codegen/nodejs/gen.go#L1878 in https://github.com/pulumi/pulumi/pull/7059 and results in `| undefined` being added to all the types.

I'm not sure yet if this is correct, so I'm going to decouple this change from the 3.5.0 k8s provider release.

**Comment by lblackstone** [comment]
Closing this in favor of #1651

**Pull request closed without merging**